File: testdata/precision-suite-objc/tp_xss.m
// XSS (WebView): an attacker-controlled comment is interpolated into an HTML
// string and loaded into a web view with -loadHTMLString:baseURL: without being
// escaped, so the markup is reflected and executed. This is CWE-79. A correct
// scanner fires TAINT-003. The safe form HTML-escapes the value first (see
// clean_safe_html.m). nox folds the selector and sees the tainted HTML reach the
// loadHTMLString sink.
#import <Foundation/Foundation.h>
#import <WebKit/WebKit.h>

void render(WKWebView *webView) {
    NSString *comment = [[[NSProcessInfo processInfo] environment] objectForKey:@"COMMENT"];
    NSString *html = [NSString stringWithFormat:@"<div>%@</div>", comment];
    [webView loadHTMLString:html baseURL:nil]; // nox-expect: TAINT-003
}

File: testdata/precision-suite-swift/tp_xss.swift
// XSS via WKWebView: an un-escaped attacker-controlled string is loaded straight
// into a web view with loadHTMLString, so any embedded markup/script executes in
// the web view's context. This is CWE-79. A correct scanner fires TAINT-003.
// nox's Swift taint model matches the tainted value flowing into loadHTMLString
// with no HTML-escape sanitizer on the path.
import Foundation
import WebKit

func render(_ webView: WKWebView) {
    let comment = CommandLine.arguments[1]
    let html = "<div>\(comment)</div>"
    webView.loadHTMLString(html, baseURL: nil) // nox-expect: TAINT-003
}
